components/homme/cmake/HommeMacros.cmake

```diff
@@ -155,7 +155,7 @@ macro(createTestExec execName execType macroNP macroNC
   ENDIF ()
 
   IF (HOMME_USE_KOKKOS)
-    TARGET_LINK_LIBRARIES(${execName} kokkos)
+    link_to_kokkos(${execName})
```

Comment:
Not other reviewers: I reorganized things a bit so that `link_to_kokkos` takes care of the library linking.